src/Compiler/NodeVisitor/AddParametersForBlockNodeVisitor.php

```diff
@@ -83,8 +88,11 @@ public function leaveNode(Node $node): ?array
                     }
                 } elseif ($defaultValue === '[]') {
                     $default = new Array_();
+                } elseif (str_contains($defaultValue, '::')) {
+                    [$className, $constantName] = explode('::', $defaultValue, 2);
+                    $default = new ClassConstFetch(new Name($className), $constantName);
                 } else {
```

Comment:
what else it could be?

Reply:
what about [1, 2, 3], or ['a', 'b', 'c']

Reply:
great idea i'll try it

Reply:
if I analyze only parameter default, I get syntax error when default is for example `[1, 2, 3, 'str']` (it contains comma and because we use comma in pattern so we destroy the array).

So I've tried to analyse whole function (`function myBlock($param1, stdClass[] $param2)`) but then I got error because `stdClass[]` is not supported be PHP so we will lost this information about array types.


Any idea how else we can handle it?

Reply:
Ha! I did it O:-) check it please